test/unit/math/rev/functor/root_finder_test.cpp

```diff
@@ -0,0 +1,327 @@
+#include <stan/math.hpp>
+#include <gtest/gtest.h>
+#include <boost/math/distributions/beta.hpp>
+#include <boost/math/special_functions/beta.hpp>
+#include <boost/math/special_functions/hypergeometric_pFq.hpp>
+#include <boost/math/special_functions/polygamma.hpp>
+#include <boost/math/special_functions/gamma.hpp>
+/*
+TEST(RevFunctor, root_finder) {
+  using stan::math::root_finder;
+  using stan::math::var;
+  // return cube root of x using 1st and 2nd derivatives and Halley.
+  // using namespace std;  // Help ADL of std functions.
+  var x = 27;
+  int exponent;
+  // Get exponent of z (ignore mantissa).
+  std::frexp(x.val(), &exponent);
+  // Rough guess is to divide the exponent by three.
+  var guess = ldexp(1., exponent / 3);
+  // Minimum possible value is half our guess.
+  var min = ldexp(0.5, exponent / 3);
+  // Maximum possible value is twice our guess.
+  var max = ldexp(2., exponent / 3);
+  // Maximum possible binary digits accuracy for type T.
+  const int digits = std::numeric_limits<double>::digits;
+  int get_digits = static_cast<int>(digits * 0.4);
+  std::uintmax_t maxit = 20;
+  auto f = [](const auto& g, const auto x) { return g * g * g - x; };
+  auto f_g = [](const auto& g, const auto x) { return 3 * g * g; };
+  auto f_gg = [](const auto& g, const auto x) { return 6 * g; };
+  var result = root_finder(std::make_tuple(f, f_g, f_gg), guess, min, max, x);
+  result.grad();
+  EXPECT_EQ(27, x.val());
+  EXPECT_NEAR(0.037037, x.adj(), 1e-6);
+}
+*/
+TEST(RevFunctor, root_finder_beta_cdf) {
+  using stan::math::var;
+  auto func = [](auto&& vals) {
+    auto p = vals(0);
+    auto alpha = vals(1);
+    auto beta = vals(2);
+    boost::math::beta_distribution<decltype(beta)> my_beta(alpha, beta);
+    return boost::math::quantile(my_beta, p);
+  };
+  double p = 0.4;
+  double a = 0.5;
+  double b = 0.5;
+  Eigen::VectorXd vals(3);
+  // p, alpha, beta
+  vals << p, a, b;
+  double fx = 0;
+  Eigen::VectorXd finit_grad_fx(3);
+  stan::math::finite_diff_gradient(func, vals, fx, finit_grad_fx, 1e-3);
+  std::cout << "--- Finit Diff----\n";
+  std::cout << "fx: " << fx;
+  std::cout << "\ngrads: \n"
+            << "p: " << finit_grad_fx(0)
+            << "\n"
+               "alpha: "
+            << finit_grad_fx(1)
+            << "\n"
+               "beta: "
+            << finit_grad_fx(2) << "\n";
```

Comment:
I understand this is here for the discussion in the PR, but in time, we'll remove the `std::cout`.